libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -2641,42 +2668,47 @@ instance HasSpec fn a => HasSpec fn [a] where
     lst <- listOfT $ genFromSpec elemS
     pureGen $ shuffle (must ++ lst)
   genFromTypeSpec ss@(ListSpec must szSpec elemS NoFold) = do
-    sz0 <- genFromSpec (sizeSpecList $ typeSpec ss)
-    let sz = sz0 - length must
+    sz0 <- genFromSizeSpec (sizeSpecList $ typeSpec ss)
+    let sz = fromIntegral (sz0 - listSize must)
     lst <-
       listOfUntilLenT
         (genFromSpec elemS)
         sz
-        ((`conformsToSpec` szSpec) . (+ length must))
+        ((`conformsToSpec` szSpec) . (+ listSize must) . fromIntegral)
```

Comment:
Well the return Integer, and the others return Int, so unless you want a bunch of 'fromIntegral' everywhere, this make things cleaner. The other thing is to use sizeOf, the function from the Sized class, but this may introduce Sized constraints.. May I'll try that and see it it happens. Might not since we usually know its a set or list.

Reply:
So it works to replace everything with sizeOf.  So I did that.